src/Snap/Snaplet/Session/Backends/RedisSession.hs

```diff
@@ -0,0 +1,245 @@
+                  Nothing -> persist (sessionKey (rsCSRFToken r'))
+                return $ (,) <$> res1 <*> res2
+              case res of
+                TxSuccess _ -> return . Payload $ S.encode r'
+                TxError e   -> error e
+                TxAborted   -> error "transaction aborted"
+          Nothing -> liftIO $ Payload . S.encode <$> mkCookieSession rng
+        setPayload mgr pl
+
+
+    --------------------------------------------------------------------------
+    --clear the session from redis and return a new empty one
+    {-reset mgr@(RedisSessionManager _ _ _ _ _ _)  = trace "RedisSessionManager reset" $ do-}
+    reset mgr@(RedisSessionManager r _ _ _ _ con)  = do
+        case r of
+          Just r' -> liftIO $
+            runRedis con $ do
+              res1 <- del [sessionKey $ rsCSRFToken r']
+              case res1 of
+                Left e  -> error $ show e
+                _ -> return ()
+          _ -> return ()
+        cs <- liftIO $ mkCookieSession (randomNumberGenerator mgr)
+        return $ mgr { session = Just cs }
+
+    --------------------------------------------------------------------------
+    touch = id
+
+    --------------------------------------------------------------------------
+    insert k v mgr@(RedisSessionManager r _ _ _ _ _) = case r of
+        Just r' -> mgr { session = Just $ modSession (HM.insert k v) r' }
+        Nothing -> mgr
+
+    --------------------------------------------------------------------------
+    lookup k (RedisSessionManager r _ _ _ _ _) = r >>= HM.lookup k . rsSession
+
+    --------------------------------------------------------------------------
+    delete k mgr@(RedisSessionManager r _ _ _ _ _) = case r of
+        Just r' -> mgr { session = Just $ modSession (HM.delete k) r' }
+        Nothing -> mgr
+
+    --------------------------------------------------------------------------
+    csrf (RedisSessionManager r _ _ _ _ _) = case r of
+        Just r' -> rsCSRFToken r'
+        Nothing -> ""
+
+    --------------------------------------------------------------------------
+    toList (RedisSessionManager r _ _ _ _ _) = case r of
+        Just r' -> HM.toList . rsSession $ r'
+        Nothing -> []
+
+------------------------------------------------------------------------------
+-- | A session payload to be stored in a SecureCookie.
+newtype Payload = Payload ByteString
+  deriving (Eq, Show, Ord, Serialize)
+
+
+------------------------------------------------------------------------------
+-- | Get the current client-side value
+getPayload :: RedisSessionManager -> Snap (Maybe Payload)
+getPayload mgr = getSecureCookie (cookieName mgr) (siteKey mgr) (timeOut mgr)
+
+
+------------------------------------------------------------------------------
+-- | Set the client-side value
+setPayload :: RedisSessionManager -> Payload -> Snap ()
+setPayload mgr = setSecureCookie
+    (cookieName mgr)
+#if MIN_VERSION_snap(1,0,0)
+    Nothing
```

Comment:
Ok, why not do the same as in https://github.com/snapframework/snap/blob/ed168734edd0b19083924d922822e9ad1314bb8a/src/Snap/Snaplet/Session/Backends/CookieSession.hs#L85 ? :)

Reply:
Good catch! Done.